### DISCUSSION TITLE
What is the reason to use the Postgres `id` column for the `_id` field in Elasticsearch?

### DISCUSSION THREAD

**Discussion by sarayourfriend** [WordPress/openverse#1947] [Q&A] [answered]
The Elasticsearch `_id` field is set to the Postgres integer ID (primary key) of the row rather than the Openverse UUID identifier:

https://github.com/WordPress/openverse/blob/697f62f01a32cb7fcf2f4a7627650a113cba40da/ingestion_server/ingestion_server/elasticsearch_models.py#L101

What is the reason for this difference? It makes certain queries against Elasticsearch oddly different (and easy to mess up, if you're not aware of this small detail), as the `_id` of the Elasticsearch document is _not_ the identifier that we use literally everywhere else in our stack.

Would it be possible to change the `_id` to be the Openverse identifier instead? What's the reason to maintain this difference, if there is one?

**Comment by obulat** [comment, marked as answer]
As far as I understand from How Indexing works diagram[^1], the `_id` was used to determine whether the items in the database need to be synced with the elasticsearch index. We have migrated to refreshing _all_ data instead of trying to sync only the latest items.

In fact, we have a [now obsolete, I think] comment in the `ingestion_server/indexer.py` describing this process:

https://github.com/WordPress/openverse/blob/4d6e995d2268dcd71381bce61b51b8eb558d7adf/ingestion_server/ingestion_server/indexer.py#L1-L6

Seems that this was the issue for implementing this kind of sync between upstream and elasticsearch: https://github.com/cc-archive/cccatalog-api/issues/7

[^1]: https://github.com/cc-archive/cccatalog-api/blob/main/ingestion_server/howitworks.png
![howitworks](https://user-images.githubusercontent.com/15233243/235424836-4e1bcd2b-48bb-4dbb-82f7-d30877549b81.png)

**Reply by sarayourfriend** [reply]
Ahhh, I see! Because the Postgres ID is an int in series, you can use it to see which one was "last", gotcha. I'll look into the code now and see how it actually works, if the database ID is indeed used in this way, and then open an issue to add documentation in the code explaining the reason for it.
